File: app/src/main/java/de/kuschku/libquassel/primitives/serializers/VoidSerializer.java
package de.kuschku.libquassel.primitives.serializers;


import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import java.io.IOException;
import java.nio.ByteBuffer;
import java.nio.channels.ByteChannel;

public class VoidSerializer implements PrimitiveSerializer<Void> {
    @NonNull
    private static final VoidSerializer serializer = new VoidSerializer();

    private VoidSerializer() {
    }

    @NonNull
    public static VoidSerializer get() {
        return serializer;
    }

    @Override
    public void serialize(@NonNull final ByteChannel channel, @NonNull final Void data) throws IOException {

    }

    @Nullable
    @Override
    public Void deserialize(@NonNull final ByteBuffer buffer) throws IOException {
        return null;
    }
}

File: app/src/main/java/de/kuschku/quasseldroid_ng/protocol/primitive/serializer/VoidSerializer.kt
package de.kuschku.quasseldroid_ng.protocol.primitive.serializer

import de.kuschku.quasseldroid_ng.protocol.Quassel_Features
import de.kuschku.quasseldroid_ng.util.nio.ChainedByteBuffer
import java.nio.ByteBuffer

object VoidSerializer : Serializer<Any?> {
  override fun serialize(buffer: ChainedByteBuffer, data: Any?, features: Quassel_Features) {
  }

  override fun deserialize(buffer: ByteBuffer, features: Quassel_Features): Any? {
    return null
  }
}
